instrumentation/opentelemetry-instrumentation-urllib/src/opentelemetry/instrumentation/urllib/__init__.py

```diff
@@ -0,0 +1,233 @@
+            "http.url": url,
+        }
+
+        with get_tracer(
+            __name__, __version__, tracer_provider
+        ).start_as_current_span(span_name, kind=SpanKind.CLIENT) as span:
+            exception = None
+            with recorder.record_client_duration(labels):
+                if span.is_recording():
+                    span.set_attribute("component", "http")
+                    span.set_attribute("http.method", method)
+                    span.set_attribute("http.url", url)
+
+                headers = get_or_create_headers()
+                propagators.inject(type(headers).__setitem__, headers)
+
+                token = context.attach(
+                    context.set_value(
+                        _SUPPRESS_URLLIB_INSTRUMENTATION_KEY, True
+                    )
+                )
+                try:
+                    result = call_wrapped()  # *** PROCEED
+                except Exception as exc:  # pylint: disable=W0703
+                    exception = exc
+                    result = getattr(exc, "file", None)
+                finally:
+                    context.detach(token)
+
+                if result is not None:
+
+                    code_ = result.getcode()
+                    labels["http.status_code"] = str(code_)
+
+                    if span.is_recording():
+                        span.set_attribute("http.status_code", code_)
+                        span.set_attribute("http.status_text", result.reason)
+                        span.set_status(
+                            Status(http_status_to_status_code(code_))
+                        )
+
+                    ver_ = str(getattr(result, "version", ""))
+                    if ver_:
+                        labels["http.flavor"] = "{}.{}".format(
+                            ver_[:1], ver_[:-1]
+                        )
+
+                if span_callback is not None:
+                    span_callback(span, result)
+
+            if exception is not None:
+                raise exception.with_traceback(exception.__traceback__)
+
+        return result
+
+    instrumented_request.opentelemetry_instrumentation_urllib_applied = True
+    OpenerDirector.open = instrumented_request
+
+
+def _uninstrument():
+    """Disables instrumentation of :code:`urllib` through this module.
+
+    Note that this only works if no other module also patches urllib."""
+    _uninstrument_from(OpenerDirector)
+
+
+def _uninstrument_from(instr_root, restore_as_bound_func=False):
+
+    instr_func_name = "open"
+    instr_func = getattr(instr_root, instr_func_name)
+    if not getattr(
+        instr_func, "opentelemetry_instrumentation_urllib_applied", False,
+    ):
+        return
+
+    original = instr_func.__wrapped__  # pylint:disable=no-member
+    if restore_as_bound_func:
+        original = types.MethodType(original, instr_root)
+    setattr(instr_root, instr_func_name, original)
+
+
+class URLLibInstrumentor(BaseInstrumentor, MetricMixin):
```

Comment:
mind moving this up to the top, below any required declarations during initialization?

It's a bit hard to review the code when the important user interface is at the bottom. I realize this is a convention for these, so if @hectorhdzg  wants to chime in that would be good too!

Reply:
Got it